### PR TITLE
docs(guide): Add Custom Backend guide

### PR DESCRIPTION
**Pull request by francoischalifour**
This is the same content as algolia/instantsearch.js#2943 with Vue code samples in "Using the search client with InstantSearch".

[:link: **Link to the guide**](https://deploy-preview-459--vue-instantsearch.netlify.com/advanced/custom-backend.html)

### CODE REVIEW

File: docs/src/advanced/custom-backend.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+> This guide is compatible with Vue InstantSearch ≥ 1.6.0.
+
+Advanced InstantSearch users may have the need to query Algolia’s servers from their backend instead of the frontend, while still being able to reuse InstantSearch widgets. Possible motivations could be for security restrictions, for SEO purposes or to enrich the data sent by the custom server (i.e. fetch Algolia data and data from their own servers). If this sounds appealing to you, feel free to follow this guide. Keep in mind though that we, at Algolia, [recommend doing frontend search](https://www.algolia.com/doc/faq/searching/searching-from-the-front-end-or-the-back-end/#we-definitely-recommend-frontend-search) for performance and high availability reasons.
+
+By the end of this guide, you will have learned how to leverage InstantSearch with your own backend architecture to query Algolia.
+
+## How it’s made possible
+
+InstantSearch is the UI part on top a search client with a state managed by a [JavaScript Helper](https://github.com/algolia/algoliasearch-helper-js). These three layers are composable and can be interchanged to leverage the InstantSearch widgets system with different search clients.
```
Comment (Haroenv) [inline docs/src/advanced/custom-backend.md]: repo link to search client?

File: docs/src/advanced/custom-backend.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+// Add the search endpoint
+app.post('/search', async (req, res) => {
+  const { requests } = req.body;
+  const results = await algoliaClient.search(requests);
+  res.status(200).send(results);
+});
+```
+
+### Supporting *Search For Facet Values*
```
Comment (Haroenv) [inline docs/src/advanced/custom-backend.md]: Vue InstantSearch (v1) doesn't have SFFV, so this section can be removed until v2

File: docs/src/advanced/custom-backend.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    }).then(res => res.json());
+  }
+};
+```
+
+We use the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) in this example to query the server; make sure to take note of the [browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#Browser_compatibility) before using it in production.
+
+> If you want to transform the data to be passed to your server, you can learn more about all the parameters that the `search()` method supports in the [Algolia API reference](https://www.algolia.com/doc/api-reference/api-parameters/?language=javascript#parameters-list).
+
+### Searching for facet values
```
Comment (Haroenv) [inline docs/src/advanced/custom-backend.md]: same as earlier

File: docs/src/advanced/custom-backend.md (hunk 1 above)
Comment (francoischalifour) [reply docs/src/advanced/custom-backend.md]: You mean this link? https://github.com/algolia/algoliasearch-client-javascript

Comment (Haroenv) [reply docs/src/advanced/custom-backend.md]: yep

Comment (francoischalifour) [reply docs/src/advanced/custom-backend.md]: I link it [at the next sentence](https://github.com/algolia/vue-instantsearch/pull/459/files#diff-541552327531ce24ddba2428315a6de3R22), which makes more sense IMO.

**Pull request merged**